RocketChat/Rocket.Chat.ReactNative
# Chore: evaluate `RoomItem`

## Proposed changes

Moves away interfaces from each component under `RoomItem` so they all are together on `RoomItem/interfaces.ts`, updates interfaces and fixes props like `avatarSize` and styles like `actionsLeftContainer`
## Issue(s)	


## How to test or reproduce

On each Item on the main screen:
1. Test swipe left or right
2. Test each option on the sides (favorite, hide, mark as read/unread)
3. Test press and long press (long press should be tested on team channels screen)
## Screenshots

## Types of changes




## Checklist



## Further comments
